src/overlays/actors/ovl_En_Hgo/z_en_hgo.c

```diff
@@ -28,54 +52,341 @@ const ActorInit En_Hgo_InitVars = {
+                    this->unk_218 = 1;
+                    func_800BDC5C(&this->skelAnime, sAnimations, 1);
+                    break;
+                case 3:
+                    this->unk_218 = 2;
+                    func_800BDC5C(&this->skelAnime, sAnimations, 2);
+                    break;
+                case 4:
+                    this->unk_218 = 3;
+                    func_800BDC5C(&this->skelAnime, sAnimations, 3);
+                    break;
+                case 5:
+                    this->unk_218 = 4;
+                    func_800BDC5C(&this->skelAnime, sAnimations, 4);
+                    break;
+                case 6:
+                    this->unk_218 = 5;
+                    func_800BDC5C(&this->skelAnime, sAnimations, 5);
+                    break;
+            }
+        } else if (Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame)) {
+            switch (this->unk_218) {
+                case 1:
+                    if ((Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame)) && (this->unk_312 == 0)) {
+                        this->unk_312 = 1;
+                        if ((gSaveContext.sceneSetupIndex == 0) &&
+                            ((globalCtx->csCtx.unk_12 == 2) || (globalCtx->csCtx.unk_12 == 4))) {
+                            Audio_PlayActorSound2(&this->actor, NA_SE_VO_GBVO02);
+                        }
+                    }
+                    break;
+                case 2:
+                    this->unk_218 = 3;
+                    func_800BDC5C(&this->skelAnime, sAnimations, 3);
+                    break;
+                case 5:
+                    this->unk_218 = 6;
+                    func_800BDC5C(&this->skelAnime, sAnimations, 6);
+            }
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Hgo/func_80BD04E0.s")
+        func_800EDF24(&this->actor, globalCtx, actionIndex);
+        return 1;
+    }
+    if ((globalCtx->csCtx.state == 0) && (((gSaveContext.weekEventReg[75]) & 0x20)) &&
+        (this->actionFunc == func_80BD0410)) {
+        this->actor.shape.rot.y = this->actor.world.rot.y;
+        Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_ELF_MSG2, this->actor.focus.pos.x, this->actor.focus.pos.y,
+                    this->actor.focus.pos.z, 7, 0, 0, 0x7F5A);
+        func_80BD0420(this);
+    }
+    this->unk_316 = 0x63;
+    return 0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Hgo/func_80BD064C.s")
+void func_80BD0B8C(EnHgo* this, GlobalContext* globalCtx) {
+    func_800E9250(globalCtx, &this->actor, &this->unk_300, &this->unk_306, this->actor.focus.pos);
+    if (this->unk_30E >= 3) {
+        this->unk_30E--;
+    } else if (this->unk_30E == 2) {
+        this->unk_30C = 1;
+        this->unk_30E = 1;
+    } else if (this->unk_30E == 1) {
+        this->unk_30C = 2;
+        this->unk_30E = 0;
+    } else {
+        this->unk_30C = 0;
+        this->unk_30E = 0x3C;
```

Comment:
```suggestion
    if (this->unk_30E > 2) {
        this->unk_30E--;
    } else if (this->unk_30E == 2) {
        this->unk_30C = 1;
        this->unk_30E = 1;
    } else if (this->unk_30E == 1) {
        this->unk_30C = 2;
        this->unk_30E = 0;
    } else {
        this->unk_30C = 0;
        this->unk_30E = 0x3C;
```
Since `2` is the max value it can take, this makes more sense to me